lib/actions/collection.js

```diff
@@ -9,24 +9,30 @@ module.exports = {
       postFetchAction,
       preFetchAction,
       trackKey,
+      adapter = defaultAdapter,
     } = options;
 
-    const storageManager = StorageManager.get();
-
     if (xhrs[trackKey] && xhrs[trackKey].state() === 'pending') xhrs[trackKey].abort();
 
-    return (dispatch) => {
+    return (dispatch, getState) => {
       if (preFetchAction) dispatch(preFetchAction);
 
-      const xhr = storageManager.storage(brainstemKey)
-        .fetch(Object.assign(fetchOptions, { remove: false }));
+      const xhr = adapter.fetchCollection(brainstemKey, {
+        dispatch,
+        getState,
+        fetchOptions,
+      });
 
-      if (postFetchAction) xhr.done(collection => dispatch(postFetchAction(collection.pluck('id'))));
+      if (postFetchAction) {
+        xhr.done(collection =>
+          dispatch(postFetchAction(adapter.collectionToIds(collection)))
+        );
+      }
 
       if (trackKey) xhrs[trackKey] = xhr;
 
       return new Promise((resolve, reject) => xhr
-        .done(collection => resolve(collection.map(model => model.attributes)))
+        .done(collection => resolve(adapter.collectionToArray(collection)))
```

Comment:
What's the use-case for overriding `collection.map`?

Reply:
We are delegating the handling of items in a collection to the adapter. Brainstem redux is now indifferent to the shape of the collection being passed in.